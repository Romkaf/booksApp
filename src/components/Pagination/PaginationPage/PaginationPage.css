.root {
	border: 1px solid var(--grey);
	border-radius: 100%;
	width: 43px;
	height: 43px;
	margin: 0 10px;
	margin-bottom: 10px;
	cursor: pointer;
	transition: box-shadow linear 0.2s;
}

.current {
	border: 2px solid var(--semi-black);
	font-weight: 700;
	cursor: auto;
	pointer-events: none;
}

.root:hover:not(.current),
.root:focus {
	box-shadow: 0 0 5pt 1.5pt var(--grey);
	outline: none;
}
