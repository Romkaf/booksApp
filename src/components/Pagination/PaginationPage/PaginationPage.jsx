import React, { memo } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import styles from './PaginationPage.css';

export const PaginationPage = ({ page, onPageClick, currentPage }) => {
	const { root, current } = styles;
	const condition = currentPage == page - 1;
	const classBtn = cx(root, { [current]: condition });

	return (
		<li>
			<button
				className={classBtn}
				aria-label="pagination-page"
				onClick={onPageClick(page)}
				tabIndex={condition ? -1 : 0}
			>
				{page}
			</button>
		</li>
	);
};

PaginationPage.propTypes = {
	page: PropTypes.number,
	onPageClick: PropTypes.func,
	currentPage: PropTypes.number,
};

export const areEqual = (prevProps, nextProps) =>
	nextProps.currentPage !== nextProps.page - 1 &&
	prevProps.currentPage !== prevProps.page - 1;

export default memo(PaginationPage, areEqual);
