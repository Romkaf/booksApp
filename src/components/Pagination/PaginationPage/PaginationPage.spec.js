import React from 'react';
import PaginationPage from './PaginationPage';
import { areEqual } from './PaginationPage';
import { shallow } from 'enzyme';

describe('PaginationPage', () => {
	const props = {
		page: 1,
		onPageClick: (it) => () => it,
		currentPage: 0,
	};

	it('should render component', () => {
		const component = shallow(<PaginationPage {...props} />);
		expect(component).toMatchSnapshot();
	});

	describe('compare function of memo', () => {
		it('should return false for current page', () => {
			const prevProps = { currentPage: 0, page: 2 };
			const nextProps = { currentPage: 1, page: 2 };
			expect(areEqual(prevProps, nextProps)).toBe(false);
		});

		it('should return false for prev current page', () => {
			const prevProps = { currentPage: 0, page: 1 };
			const nextProps = { currentPage: 1, page: 1 };
			expect(areEqual(prevProps, nextProps)).toBe(false);
		});
	});
});
