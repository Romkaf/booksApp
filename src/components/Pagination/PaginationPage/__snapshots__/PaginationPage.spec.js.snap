// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationPage should render component 1`] = `
<li>
  <button
    aria-label="pagination-page"
    className="root current"
    onClick={[Function]}
    tabIndex={-1}
  >
    1
  </button>
</li>
`;
